Graphql integration (#525)
DataDog/dd-sdk-reactnative: packages/react-native-apollo-client/README.md

import { DatadogLink } from '@datadog/mobile-react-native-apollo-client';

const apolloClient = new ApolloClient({
    link: from([
        new DatadogLink(),
        new HttpLink({ uri: 'https://my.api.com/graphql' }) // always in last position
    ])
});
```

For more information on Apollo Client Links, refer to the [official documentation][4].

### Removing GraphQL information

Use a `resourceEventMapper` in your Datadog configuration to remove sensitive data from GraphQL variables:

```javascript
const datadogConfiguration = new DatadogProviderConfiguration(
    '<CLIENT_TOKEN>',
    '<ENVIRONMENT_NAME>',
    '<RUM_APPLICATION_ID>',
    true,
    true,
    true
);

datadogConfiguration.resourceEventMapper = event => {
    // Variables are stored in event.context['_dd.graphql.variables'] as a JSON string when present
    if (event.context['_dd.graphql.variables']) {
        const variables = JSON.parse(event.context['_dd.graphql.variables']);
        if (variables.password) {
            variables.password = '***';
        }
        event.context['_dd.graphql.variables'] = JSON.stringify(variables);
    }

    return event;
};
```

> ## Access your GraphQL resources information

Reviewer: Do you anticipate customers asking about this? Recommend removing this section.

Author: Right now adding the GraphQL integration will change strictly nothing for users as the web platform does not display the information yet.

The goal is to warn users that this is not usable yet in case they came across this package so they don't try it out and feel let down. 

I can replace this by a warning at the top of the README if that makes the purpose clearer?

Reviewer: @louiszawadzki If this is a feature that we plan on supporting in the future, an alert it might work better. Not sure if the alert div is supported in this repo, but this is what we usually add to highlight information in docs:
```suggestion
<div class="alert alert-info">Adding the GraphQL integration does not change any existing features. The information is not surfaced on the Datadog platform.</div>
```

# React-Native Monitoring for @apollo/client

## Overview

> Real User Monitoring (RUM) allows you to monitor your GraphQL resources.

Reviewer: ```suggestion
Monitor your GraphQL resources with Real User Monitoring (RUM) and perform the following:
```

> You can perform the following:

Reviewer: ```suggestion
```

-   Identify GraphQL queries and mutations
-   Identify GraphQL variables used in queries and mutations

RUM supports GraphQL requests created using [@apollo/client][2].

## Setup

### Prerequisites

> Set up the RUM Browser SDK on the web page you want rendered on your mobile React Native application. For more information, see [RUM Browser Monitoring][1].

Reviewer: ```suggestion
Set up the RUM Browser SDK on the web page you want to render on the mobile React Native application. For more information, see [RUM Browser Monitoring][1].
```
